esp-hal/src/dma/mod.rs

```diff
@@ -1840,6 +1878,519 @@ where
     }
 }
 
+/// Error returned from Dma[Tx|Rx|TxRx]Buf operations.
+#[derive(Debug)]
+pub enum DmaBufError {
+    /// More descriptors are needed for the buffer size
+    InsufficientDescriptors,
+    /// Descriptors or buffers are not located in a supported memory region
+    UnsupportedMemoryRegion,
+}
+
+/// DMA transmit buffer
+///
+/// This is a contiguous buffer linked together by DMA descriptors of length
+/// 4092. It can only be used for transmitting data to a peripheral's FIFO.
+/// See [DmaRxBuf] for receiving data.
+#[derive(Debug)]
+pub struct DmaTxBuf {
+    descriptors: &'static mut [DmaDescriptor],
+    buffer: &'static mut [u8],
+}
+
+impl DmaTxBuf {
+    /// Creates a new [DmaTxBuf] from some descriptors and a buffer.
+    ///
+    /// There must be enough descriptors for the provided buffer.
+    /// Each descriptor can handle 4092 bytes worth of buffer.
+    ///
+    /// Both the descriptors and buffer must be in DMA-capable memory.
+    /// Only DRAM is supported.
+    pub fn new(
+        descriptors: &'static mut [DmaDescriptor],
+        buffer: &'static mut [u8],
```

Comment:
Could this be something that implements WriteBuffer instead?  For instance  A frame buffer that us made from a `&'static mut [u16]` or event `&'static mut [Entry]`.

Reply:
`embedded_dma` seems to have been abandoned, just my 5 cents

Reply:
As one of the further enhancements I'm planning to make, I'm going to make a trait for these buffer objects and this will make it easier to throw in customisations like this.

(This will likely be the next PR I make)

I want to keep this PR simple for now. The merge conflicts are already killing my motivation.

Reply:
Yes but these were re-implemented as `esp_hal::dma::ReadBuffer` and `esp_hal::dma::WriteBuffer`

Reply:
Let me know if I can help - I have a driver for HUB75 displays where I Implement `ReadBuffer`.